app/routes/ap-ar-report-listing.js

```diff
@@ -0,0 +1,109 @@
+const config = require('../config/storage')
+
+function generateRoutes (reportName, reportDataUrl, reportDataKey) {
+  return [
+    {
+      method: 'GET',
+      path: `/report-list/${reportName}`,
+      options: {
+        auth: { scope: [holdAdmin, schemeAdmin, dataView] },
+        handler: async (request, h) => {
+          return h.view(`reports-list/${reportName}`)
+        }
+      }
+    },
+    {
+      method: 'GET',
+      path: `/report-list/${reportName}/download`,
+      options: {
+        auth: { scope: [holdAdmin, schemeAdmin, dataView] },
+        validate: {
+          query: apListingSchema,
+          failAction: async (request, h, err) => {
+            request.log(['error', 'validation'], err)
+            const errors = err.details
+              ? err.details.map(detail => {
+                  return {
+                    text: detail.message,
+                    href: '#' + detail.path[0]
+                  }
+                })
+              : []
+            const data = { errors }
+            return h.view(`reports-list/${reportName}`, data).code(400).takeover()
+          }
+        },
+        handler: async (request, h) => {
+          const { 'start-date-day': startDay, 'start-date-month': startMonth, 'start-date-year': startYear, 'end-date-day': endDay, 'end-date-month': endMonth, 'end-date-year': endYear } = request.query
+
+          let url = reportDataUrl
+          let startDate, endDate
+
+          if (startDay && startMonth && startYear) {
+            startDate = `${startYear}-${String(startMonth).padStart(2, '0')}-${String(startDay).padStart(2, '0')}`
+          } else {
+            startDate = '2015-01-01'
+          }
+
+          if (endDay && endMonth && endYear) {
+            endDate = `${endYear}-${String(endMonth).padStart(2, '0')}-${String(endDay).padStart(2, '0')}`
+          } else if (startDate) {
+            const now = new Date()
+            endDate = `${now.getFullYear()}-${String(now.getMonth() + 1).padStart(2, '0')}-${String(now.getDate()).padStart(2, '0')}`
+          }
+
+          if (startDate && endDate) {
+            url += `?startDate=${startDate}&endDate=${endDate}`
+          }
+
+          try {
+            const response = await api.getTrackingData(url)
+            const trackingData = response.payload
+
+            const selectedData = trackingData[reportDataKey].map(data => {
+              return {
+                Filename: data.batch,
+                'Date Time': data.lastUpdated,
+                Event: data.status,
+                FRN: data.frn,
+                'Original Invoice Number': data.originalInvoiceNumber,
+                'Original Invoice Value': data.value,
+                'Invoice Number': data.invoiceNumber,
+                'Invoice Delta Amount': data.deltaAmount,
+                'D365 Invoice Imported': data.routedToRequestEditor,
+                'D365 Invoice Payment': data.settledValue,
```

Comment:
Can we add a step to remove this D365 Invoice Payment for the AR report? We don't get settlement data back for those so will always be null.